src/Microsoft.Health.Dicom.Core/Extensions/DicomDatasetExtensions.cs

```diff
@@ -158,5 +159,36 @@ public static void AddValueIfNotNull<T>(this DicomDataset dicomDataset, DicomTag
                 dicomDataset.Add(dicomTag, value);
             }
         }
+
+        /// <summary>
+        /// Get custom tag indexes
+        /// </summary>
+        /// <param name="dicomDataset">The dicom dataset.</param>
+        /// <param name="customTagStoreEntries">The custom tag entries stored.</param>
+        /// <returns>The indexes.</returns>
+        public static Dictionary<long, DicomItem> GetCustomTagIndexes(this DicomDataset dicomDataset, IDictionary<string, CustomTagStoreEntry> customTagStoreEntries)
```

Comment:
Why would we not convert the path to DicomTag and use GetSingleValueOrDefault instead?

Reply:
hi Smitha, I'm not getting this.

When we support embed custom tag, path could be a list of DicomTag.